src/script/api/script_event_types.cpp

```diff
@@ -118,23 +118,18 @@ bool ScriptEventCompanyAskMerger::AcceptMerger()
 	return ScriptObject::DoCommand(0, this->owner, 0, CMD_BUY_COMPANY);
 }
 
-ScriptEventAdminPort::ScriptEventAdminPort(const char *json) :
+ScriptEventAdminPort::ScriptEventAdminPort(const std::string &json) :
 		ScriptEvent(ET_ADMIN_PORT),
-		json(stredup(json))
+		json(json)
 {
 }
 
-ScriptEventAdminPort::~ScriptEventAdminPort()
-{
-	free(this->json);
-}
-
 #define SKIP_EMPTY(p) while (*(p) == ' ' || *(p) == '\n' || *(p) == '\r') (p)++;
 #define RETURN_ERROR(stack) { ScriptLog::Error("Received invalid JSON data from AdminPort."); if (stack != 0) sq_pop(vm, stack); return nullptr; }
 
 SQInteger ScriptEventAdminPort::GetObject(HSQUIRRELVM vm)
 {
-	char *p = this->json;
+	char *p = this->json.data();
```

Comment:
Afaict, the only reason this is mutable is so that ReadString below can do dirty things to make splitting strings slightly more efficient. Perhaps there's a better solution (read with length function somewhere?) or just copying the string.

Reply:
It should probably use std::string_view for the dirty things, however that makes reviewing this PR significantly more difficult and as a result quickly conflicts with other changes for the network code. So I chose to split the straight forward stuff that messes with lots of places from the more complicated localized things.
For example 9276 was first part of my 9263 PR, but 9272 and this also depended on that. And I'm not quite sure everyone is happy with 9276 as it is, so that would have postponed the other PRs as well making me have to rebase and solve conflicts more than I want to.

Reply:
Looking at the code I can quite easily make `char *p` to `const char *p` as the messing with the string is not even needed at all.